priv/templates/phx.gen.auth/auth.ex

```diff
@@ -42,15 +43,21 @@ defmodule <%= inspect auth_module %> do
   the existing remember_me setting is kept, writing a new remember_me cookie.
   """
   def log_in_<%= schema.singular %>(conn, <%= schema.singular %>, params \\ %{}) do
-    token = <%= inspect context.alias %>.generate_<%= schema.singular %>_session_token(<%= schema.singular %>)
     <%= schema.singular %>_return_to = get_session(conn, :<%= schema.singular %>_return_to)
+
+    conn
+    |> create_or_extend_session(<%= schema.singular %>, params)
+    |> redirect(to: <%= schema.singular %>_return_to || signed_in_path(conn))
+  end
+
+  defp create_or_extend_session(conn, <%= schema.singular %>, params) do
```

Comment:
@josevalim as we don't use the variant without redirect outside of this module, I decided to make it a private function instead.